File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1;
        int maxArea = INT_MIN;
        while (left < right) {
            int area = (right - left) * min(height[left], height[right]);
            maxArea = max(maxArea, area);
            if (height[left] < height[right]) {
                left++;
            } else if (height[right] < height[left]) {
                right--;
            } else {
                left++;
            }
        }
        return maxArea;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public static int maxArea(int[] heights) {
        int left = 0, right = heights.length-1;
        int maxArea = 0;
        while (left < right) {
            int minH = Math.min(heights[left], heights[right]);
            int area = minH * (right-left);
            maxArea = Math.max(maxArea, area);
            if (heights[left] < heights[right]) left++;
            else right--;
        }
        return maxArea;
    }
}